internal/internal_task_handlers.go

```diff
@@ -877,7 +883,7 @@ ProcessEvents:
 				if err != nil {
 					return nil, err
 				}
-				if w.isWorkflowCompleted {
+				if w.isWorkflowCompleted && !shouldForceReplayCheck() {
```

Comment:
not sure what you mean by a "completed command" can you elaborate? 

The problem with breaking out is described here https://github.com/temporalio/sdk-go/issues/983, basically if we break we skip populating the `replayCommands` list and that is a important part of checking if a replay was deterministic.

Reply:
Hrmm, ok. I was expecting that if we skip populating replay commands then this workflow would add a workflow completed command and inside the replayer that would mismatch with the next event not being a complete command. But I think this works too. And we're pretty sure it's isolated only to the replayer.

But if this change is only for replayer and only applies to a completed workflow, I think we're good.

Reply:
The break would also causes false negatives if activities are removed from a workflow because we would break before processing the event. I made a test for this called `TestBadEmptyWorkflow` that is an extreme case.

Reply:
> I was expecting that if we skip populating replay commands then this workflow would add a workflow completed command and inside the replayer that would mismatch with the next event not being a complete command

Ah here is the thing, where would it mismatch with the next event? In the replayer (I assume you mean `WorkflowReplayer `) we only check the last event in the history is a workflow completed event